pkg/wakaama/patches/0015-fixed-alignment-problem-in-lwm2m_data_t.patch

```diff
@@ -0,0 +1,32 @@
+From 5d0fd8783b734e8adeec3b9393e84dc8f4f7eb1e Mon Sep 17 00:00:00 2001
+From: Christian Manal <moenoel@informatik.uni-bremen.de>
+Date: Wed, 16 May 2018 11:30:02 +0200
+Subject: [PATCH] fixed alignment problem in lwm2m_data_t
+
+---
+ core/liblwm2m.h | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/core/liblwm2m.h b/core/liblwm2m.h
+index 3f16a6d..53616bf 100644
+--- a/core/liblwm2m.h
++++ b/core/liblwm2m.h
+@@ -299,7 +299,6 @@ typedef struct _lwm2m_data_t lwm2m_data_t;
+ 
+ struct _lwm2m_data_t
+ {
+-    lwm2m_data_type_t type;
```

Comment:
@moenoel is this safe? E.g wakaama code could eventually cast `_lwm2m_data_t` to `lwm2m_data_type_t` because address_of(_lwm2m_data_t) == addess_of(_lwm2m_data_t->type). With this change, address_of(_lwm2m_data_t) == address_of(_lwm2m_data_t->id).

Do you remember the reason behind this?

Reply:
Not entirely sure if it's safe, though I didn't encounter any obvious problems running it. The reason for that change was to get rid of a memory alignment issue on the ARM boards we used. 

IIRC `lwm2m_data_type_t` is an enum and apparently enums can have strange widths, so all the following struct members got misaligned, leading to hard faults. Making the enum member the last one fixed that for us.